eval/src/vespa/eval/eval/tensor_instructions.h

```diff
@@ -0,0 +1,24 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "interpreted_function.h"
+
+namespace vespalib {
+
+class Stash;
+
+namespace eval {
+
+struct JoinPlan;
+
+namespace tensor_instruction {
+
+using join_fun_t = double (*)(double, double);
+
+InterpretedFunction::Instruction make_join(const ValueType &lhs_type, const ValueType &rhs_type, join_fun_t function,
+                                           const ValueBuilderFactory &factory, Stash &stash);
+
+}
+}
```

Comment:
ending three nested namespaces like this feels confusing to me.
I suggest something like:

```
namespace vespalib { class Stash; }

namespace vespalib::eval::tensor_instruction {
... rest of file ...
}
```

Reply:
agree